rxjava2/src/main/java/com/linecorp/armeria/common/rxjava2/RequestContextAssembly.java

```diff
@@ -91,161 +85,120 @@ public static synchronized void enable() {
             return;
         }
 
-        oldOnObservableAssembly = RxJavaPlugins.getOnObservableAssembly();
-        RxJavaPlugins.setOnObservableAssembly(compose(
-                oldOnObservableAssembly,
-                new ConditionalOnCurrentRequestContextFunction<Observable>() {
+        oldOnObservableSubscribe = RxJavaPlugins.getOnObservableSubscribe();
+        RxJavaPlugins.setOnObservableSubscribe(biCompose(
+                oldOnObservableSubscribe,
+                new BiConditionalOnCurrentRequestContextFunction<Observable, Observer>() {
                     @Override
-                    Observable applyActual(Observable o, RequestContext ctx) {
-                        if (!(o instanceof Callable)) {
-                            return new RequestContextObservable(o, ctx);
+                    Observer applyActual(Observable observable, Observer observer, RequestContext ctx) {
+                        try (SafeCloseable ignored = ctx.push()) {
+                            return new RequestContextObserver<>(observer, ctx);
                         }
```

Comment:
Don't we need to push a `RequestContext` while creating a `RequestContextObserver`?